### ISSUE TITLE
[ROG Ally X] List of issues

### ISSUE THREAD

**Issue by Revival8697** [CachyOS/CachyOS-Handheld#33]
- The controller stops working after waking up from sleep (either through the Steam menu or power button) and requires an [EC reset](https://www.asus.com/support/faq/1050239) to function again..
- Adaptive brightness does not work.
- No TDP/fan control (Can be somewhat configured with https://aur.archlinux.org/packages/rog-control-center).
- No RGB control.
- Mouse emulation though joystick speed is 275% by default (why?), should be 100%.

Touch screen unfriendly stuff:
- Display scale keep resetting to 125% after rebooting, 150% looks good to me.
- The default on screen keyboard doesn't work on all apps, Steam's should be used instead (Left joystick click).
- KDE's title buttons are Medium, hard to tap on, should be Very Large instead.
- Cursor size should be bigger, 36 looks good to me.
- Task bar width should be bigger, 56 (at 125%) looks good to me.

**Comment by ptr1337** [comment]
Since you have already a bunch of points, feel free to PR them.
Nekoh is currently heavily for his study, and therefore is less maintenance.

**Comment by Revival8697** [comment]
> Since you have already a bunch of points, feel free to PR them.

For desktop stuff, I'm not familiar with plasma config files but I will look into it.

**Comment by 1Naim** [comment]
> * VDPAU is not configured (https://wiki.archlinux.org/title/Hardware_video_acceleration#Configuring_VDPAU).

Can you provide the output of `vdpauinfo` Install it from the repositories if you do not have it installed.

**Comment by Revival8697** [comment]
> Can you provide the output of `vdpauinfo` Install it from the repositories if you do not have it installed.

Hey, my device is in for warranty (peak ASUS QC). I don't remember what the error was but I had to manually set `VDPAU_DRIVER=radeonsi` for it to work.

**Comment by 1Naim** [comment]
> > Can you provide the output of `vdpauinfo` Install it from the repositories if you do not have it installed.
> 
> Hey, my device is in for warranty (peak ASUS QC). I don't remember what the error was but I had to manually set `VDPAU_DRIVER=radeonsi` for it to work.

This usually happens if there are more than one vdpau drivers installed on the system. AFAIK the Ally X should only have one GPU and therefore only need the radeonsi vdpau driver. I'll check with other Ally X users if this is needed. If so then this should be handled by chwd.

**Comment by En3Tho** [comment]
One thing I'm really missing is Gyro support.

@ptr1337 honestly I'm a total newb but maybe I can gather some information so other more knowledgeable people can step in too. Where do I even start? 

Is it kernel/driver/script or all three? For example does gyro support come from cachyos build/packages or it is "deck" packages thingy and they have to be some differently/properly configured?

**Comment by Revival8697** [comment]
@1Naim Since ASUS has non-existent QC, I have decided to sell my device once I get it back from the 5th warranty repair.

Please close this if most issues are addressed since I won't be maintaining it any further.

**Comment by 1Naim** [comment]
> * Frame Limit doesn't work when VRR is enabled

This should work after https://github.com/CachyOS/CachyOS-PKGBUILDS/pull/584

> * VDPAU is not usable OoTB (https://wiki.archlinux.org/title/Hardware_video_acceleration#Configuring_VDPAU).

I still don't understand why this is a problem. Looking at the spec sheet, Ally X doesn't seem to be a hybrid device so there should be no need to explicitly set the env var.

**Comment by Revival8697** [comment]
> I still don't understand why this is a problem. Looking at the spec sheet, Ally X doesn't seem to be a hybrid device so there should be no need to explicitly set the env var.

Hey there, I got a Steam Deck. Expect a list of issues for it as soon as I get a keyboard. 😉

Anyway, I have the same issue with VDPAU, seems like I can blame Nvidia for yet another thing.

![Image](https://github.com/user-attachments/assets/6d96bf0b-c6f4-44c5-84f9-312a51be761e)

**Comment by 1Naim** [comment]
> Anyway, I have the same issue with VDPAU, seems like I can blame Nvidia for yet another thing.

Yeah, it's loading NVIDIA by default for me too. Do you know of any real applications that use vdpau? `vdpau` doesn't seem to be trying to load any other library besides NVIDIA's on failure. I'm asking to see if applications that actually use this use the correct driver instead of what `vdpauinfo` is doing.

**Comment by AutomaticFailure** [comment]
Unsure if I should open a new issue or not. Updated system last week and again this morning and the back M1/M2 buttons are unable to be used. Unable to remap them as if they are not even detected. They did work previously from a week ago. Unsure of how to rectify it.
